src/main/java/com/numble/karrotmarket/auth/service/CustomUserDetailsService.java

```diff
@@ -0,0 +1,36 @@
+package com.numble.karrotmarket.auth.service;
+
+import com.numble.karrotmarket.common.exception.error.ErrorCode;
+import com.numble.karrotmarket.user.domain.User;
+import com.numble.karrotmarket.user.repository.UserRepository;
+import java.util.Collections;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class CustomUserDetailsService implements UserDetailsService {
+
+    private final UserRepository userRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(final String userPk) throws UsernameNotFoundException {
+        return userRepository.findById(Long.parseLong(userPk))
+            .map(this::createUserDetails)
```

Comment:
밑에 있는 코드처럼 simply하게 사용할 수 있지 않나요?

```suggestion
            .map(createUserDetails)
```

Reply:
.map은 Function을 매개변수로 받기 때문에 아래 코드로 작성됩니다.
`.map(i -> createUserDetails(i))`

위 방법보다는 매개 변수 생략이 가능한  메소드 참조 방식이 더 심플하여 사용했습니다.